### PR TITLE
make wire protocol message size configurable.

### PR DESCRIPTION
**Pull request by raulk**
Relates to #209, but doesn't close it because this implements a configurable global maximum message size.

This PR introduces the `WithMaxMessageSize(size int)` functi onal option that can be used in the pubsub constructor. It also introduces a unit test, and detailed godocs for this feature.

### CODE REVIEW

**Comment by raulk** [comment]
cc @vishalchangraniaxiom

File: go.mod [hunk 1]
```diff
@@ -2,7 +2,7 @@ module github.com/libp2p/go-libp2p-pubsub
 
 require (
 	github.com/gogo/protobuf v1.3.1
-	github.com/hashicorp/golang-lru v0.5.4
+	github.com/hashicorp/golang-lru v0.5.4 // indirect
```
Comment (aschmahmann) [inline go.mod:5]: Is this an artifact from your local branch or do we need to change the dependency to indirect anyway?

File: floodsub_test.go [hunk 2]
```diff
@@ -1119,3 +1119,46 @@ func TestMessageSender(t *testing.T) {
...
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	hosts := getNetHosts(t, ctx, 10)
+
+	// use a 4mb limit; default is 1mb; we'll test with a 2mb payload.
+	psubs := getPubsubs(ctx, hosts, WithMaxMessageSize(1<<22))
+
+	sparseConnect(t, hosts)
+	time.Sleep(time.Millisecond * 100)
```
Comment (aschmahmann) [inline floodsub_test.go:1133]: These make me sad and they're all over this code base. While we could certainly improve the API (e.g. by allowing a Readiness check even without a dummy discovery mechanism) we can already remove the sleep by following the pattern in this test.

https://github.com/libp2p/go-libp2p-pubsub/blob/a1999db8934dc4320aaaa578aed05873113f1a04/topic_test.go#L122

https://github.com/libp2p/go-libp2p-pubsub/blob/a1999db8934dc4320aaaa578aed05873113f1a04/topic_test.go#L145

**Review by aschmahmann — APPROVED** [review]
LGTM, although I'd like to give @vyzo an opportunity to verify he's ok with this change. This seems to be a solution to a request made at https://github.com/libp2p/specs/issues/118#issuecomment-560607776

**Review by Stebalien — APPROVED** [review]

**Comment by raulk** [comment]
Enough approvals; merging.

**Pull request merged**